lms/static/js/learner_dashboard/views/course_entitlement_view.js

```diff
@@ -0,0 +1,395 @@
+                     if (currentSessionId === newSessionId) {
+                         enrollBtnInitial.addClass('disabled');
+                         this.removeDialog(enrollBtnInitial);
+                         return;
+                     }
+                     enrollBtnInitial.removeClass('disabled');
+
+                     // Update button text specifying if the user is initially enrolling, changing or leaving a session.
+                     if (newSessionId) {
+                         enrollText = currentSessionId ? gettext('Change Session') : gettext('Select Session');
+                     } else {
+                         enrollText = gettext('Leave Current Session');
+                     }
+                     this.$('.enroll-btn-initial').text(enrollText);
+                     this.removeDialog($('.enroll-btn-initial'));
+                     this.initializeVerificationDialog($('.enroll-btn-initial'));
+                 },
+
+                 toggleSessionSelectionPanel: function() {
+                     /*
+                     Opens and closes the session selection panel.
+                     */
+                     this.$el.toggleClass('hidden');
+                     if (!this.$el.hasClass('hidden')) {
+                         // Set focus to the session selection for a11y purposes
+                         this.$('.session-select').focus();
+                         this.$('.enroll-btn-initial').popover('hide');
+                     }
+                     this.updateEnrollBtn();
+                 },
+
+                 initializeVerificationDialog: function(invokingElement) {
+                     /*
+                     Instantiates an instance of the Bootstrap v4 dialog modal and attaches it to the passed in element.
+
+                     This dialog acts as the second step in verifying the user's action to select, change or leave an
+                     available course session.
+                      */
+                     var confirmationMsgTitle,
+                         confirmationMsgBody,
+                         popoverDialogHtml,
+                         currentSessionId = this.entitlementModel.get('currentSessionId'),
+                         newSessionId = this.$('.session-select').find('option:selected').data('session_id');
+
+                     // Update the button popover text to enable two step authentication.
+                     if (newSessionId) {
+                         confirmationMsgTitle = !currentSessionId ?
+                             gettext('Are you sure that you would like to select this session?') :
+                             gettext('Are you sure that you would like to switch session?');
+                         confirmationMsgBody = !currentSessionId ? '' :
+                             gettext('Please know that by choosing to switch session you will' +
+                                 ' lose any progress you have made in this session.');
```

Comment:
I'm not sure how the extraction scripts will handle the `+` here. It might be better to just have it be one long line.